src/HDF5Msg.chpl

```diff
@@ -896,6 +898,371 @@ module HDF5Msg {
+                        var valEntry:SymEntry = toSymEntry(toGenSymEntry(vals), real);
+
+                        //localize values and write dataset
+                        var localVals = new lowLevelLocalizingSlice(valEntry.a, 0..#valEntry.size);
+                        var val_dims: C_HDF5.hsize_t = valEntry.size:C_HDF5.hsize_t;
+                        C_HDF5.H5LTmake_dataset(file_id, "/%s/values".format(group).c_str(), 1:c_int, val_dims, getHDF5Type(real), localVals.ptr);
+                        
+                        //localize segments and write dataset
+                        var localSegs = new lowLevelLocalizingSlice(segEntry.a, 0..#segEntry.size);
+                        var seg_dims: C_HDF5.hsize_t = segEntry.size:C_HDF5.hsize_t;
+                        C_HDF5.H5LTmake_dataset(file_id, "/%s/segments".format(group).c_str(), 1:c_int, seg_dims, getHDF5Type(int), localSegs.ptr);
+
+                        writeArkoudaMetaData(file_id, group, objType, getHDF5Type(real));
+                    } when (DType.Bool) {
+                        var valEntry:SymEntry = toSymEntry(toGenSymEntry(vals), bool);
+
+                        //localize values and write dataset
+                        var localVals = new lowLevelLocalizingSlice(valEntry.a, 0..#valEntry.size);
+                        var val_dims: C_HDF5.hsize_t = valEntry.size:C_HDF5.hsize_t;
+                        C_HDF5.H5LTmake_dataset(file_id, "/%s/values".format(group).c_str(), 1:c_int, val_dims, getDataType(bool), localVals.ptr);
+                        
+                        //localize segments and write dataset
+                        var localSegs = new lowLevelLocalizingSlice(segEntry.a, 0..#segEntry.size);
+                        var seg_dims: C_HDF5.hsize_t = segEntry.size:C_HDF5.hsize_t;
+                        C_HDF5.H5LTmake_dataset(file_id, "/%s/segments".format(group).c_str(), 1:c_int, seg_dims, getDataType(int), localSegs.ptr);
+
+                        writeArkoudaMetaData(file_id, group, objType, getDataType(bool));
+                        writeArkoudaMetaData(file_id, "%s/segments".format(group), "pdarray", getDataType(int));
+                        writeArkoudaMetaData(file_id, "%s/values".format(group), "pdarray", getDataType(bool));
+                    }
+                    otherwise {
+                        throw getErrorWithContext(
+                           msg="Unsupported SegArray DType %s".format(dtype2str(dType)),
+                           lineNumber=getLineNumber(),
+                           routineName=getRoutineName(), 
+                           moduleName=getModuleName(),
+                           errorClass="IllegalArgumentError");
+                    }
+                }
+                
+                C_HDF5.H5Fclose(file_id);
+            }
+            when MULTI_FILE {
+                var filenames = prepFiles(filename, mode, segEntry.a);
+
+                ref se = segEntry;
```

Comment:
I don't think the ref here is really getting anything since it is only used on the next line.

Reply:
Removed